src/main/java/org/spongepowered/common/mixin/core/world/gen/populators/MixinWorldGenDungeons.java

```diff
@@ -24,44 +24,68 @@
  */
 package org.spongepowered.common.mixin.core.world.gen.populators;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+
 import com.flowpowered.math.vector.Vector3i;
 import com.google.common.base.Objects;
+import net.minecraft.tileentity.MobSpawnerBaseLogic;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 import net.minecraft.world.gen.feature.WorldGenDungeons;
 import net.minecraft.world.gen.feature.WorldGenerator;
 import org.spongepowered.api.data.manipulator.mutable.MobSpawnerData;
+import org.spongepowered.api.entity.EntityArchetype;
 import org.spongepowered.api.item.inventory.ItemStackSnapshot;
+import org.spongepowered.api.util.weighted.ImmutableProxyWeightedTable;
 import org.spongepowered.api.util.weighted.LootTable;
 import org.spongepowered.api.util.weighted.VariableAmount;
+import org.spongepowered.api.util.weighted.WeightedSerializableObject;
+import org.spongepowered.api.util.weighted.WeightedTable;
 import org.spongepowered.api.world.extent.Extent;
 import org.spongepowered.api.world.gen.PopulatorType;
 import org.spongepowered.api.world.gen.PopulatorTypes;
 import org.spongepowered.api.world.gen.populator.Dungeon;
+import org.spongepowered.asm.mixin.Final;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.Shadow;
 import org.spongepowered.asm.mixin.injection.At;
 import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
-import org.spongepowered.common.interfaces.world.gen.IWorldGenDungeons;
+import org.spongepowered.common.data.processor.common.SpawnerUtils;
+import org.spongepowered.common.entity.EntityUtil;
 
+import java.util.Optional;
 import java.util.Random;
 
+import javax.annotation.Nullable;
+
 @Mixin(WorldGenDungeons.class)
-public abstract class MixinWorldGenDungeons extends WorldGenerator implements Dungeon, IWorldGenDungeons {
+public abstract class MixinWorldGenDungeons extends WorldGenerator implements Dungeon {
+
+    // To allow Forge's DungeonHooks to be added to DefaultEntities list
+    public static final WeightedTable<EntityArchetype> DEFAULT_ENTITIES_INTERNAL = new WeightedTable<>();
+    // We force this one to be immutable to avoid any wacky changes to it from plugins
+    public static final WeightedTable<EntityArchetype> DEFAULT_ENTITIES = new ImmutableProxyWeightedTable<>(DEFAULT_ENTITIES_INTERNAL);
```

Comment:
I'm not certain if this is OK, or whether I should create some kind of registry to put all this stuff in